doc/workshops/financial.rst

```diff
@@ -2,4 +2,1065 @@
+
+Let's put all this into a function:
+
+.. code-block:: python
+
+   def portfolio_return(returns, weights):
+       return np.dot(returns.mean(), weights) * 252
+
+Let's look at the expected return of this portfolio if we hold it for one year:
+
+.. code-block:: python
+
+   portfolio_return(rets_1, weights_1)
+
+   # 0.2859066023606343
+
+An expected gain of almost 30% in one year. Not bad, right? But don't forget the other side of this coin: volatility. This calculation is a bit more complex. First, the dot product of the annualized covariance of the returns (this is multiplied by the number of trading days in a year) and the weights is calculated. Then the dot product of the weights and the previous result is obtained. Finally, the square root of this result is extracted. Let's implement this into a function as well.
+
+.. code-block:: python
+
+   def portfolio_volatility(returns, weights):
+       return np.dot(weights, np.dot(returns.cov() * 252, weights)) ** 0.5
+
+Let's see how our portfolio performs with respect to its volatility:
+
+.. code-block:: python
+
+   portfolio_volatility(rets_1, weights_1)
+
+   # 0.23704031354688784
+
+If high return is desirable, high volatility is undesirable. The risk of this portfolio is relatively large.
+
+
+Sharpe ratio
+~~~~~~~~~~~~
+
+The `Sharpe`_ ratio or index is a measure of portfolio performance. It relates the portfolio's return to its volatility, comparing the expected/realized return with the expected/realized risk. It is calculated as the difference between the actual investment returns and the expected return in a zero-risk situation, divided by the volatility of the investment. **It provides a model of the additional amount of returns received for each additional unit of risk**.
+
+.. _Sharpe: https://en.wikipedia.org/wiki/Sharpe_ratio
+
+Let's formalize this in a function:
+
+.. code-block:: python
+
+   def portfolio_sharpe(returns, weights):
+       return portfolio_return(returns, weights) / portfolio_volatility(returns, weights)
+
+And let's apply this to our portfolio.
+
+.. code-block:: python
+
+   portfolio_sharpe(rets_1, weights_1)
+
+   # 1.2061518063427656
+
+.. important::
+   
+   The Sharpe ratio measure is best understood in context: when comparing two or more portfolios, the one with the higher Sharpe ratio provides more profit for the same amount of risk.
+
+We can also use a Monte Carlo simulation to randomize the weights of each stock in the portfolio so that we can see the range over which the Sharpe ratio can vary. In this way we can plot some scenarios that together will give us a good insight into of the relationship between expected returns and expected volatility.
```

Comment:
```suggestion
We can also use a Monte Carlo simulation to randomize the weights of each stock in the portfolio so that we can see the range over which the Sharpe ratio can vary. In this way we can plot some scenarios that together will give us a good insight of the relationship between expected returns and expected volatility.
```